### PR TITLE
Installation under CLI doesn't take BASE_URI and Apache rewrite in consideration

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | File `index_cli.php` ignores the `__PS_BASE_URI__` const when running under a CLI context. The problem is `Datas` class needs `Validate` class, `Validate` class need to be autoloaded, and to make the autoload work, it need `__PS_BASE_URI__`...  <br>To make it work, I add a `class_exist` check in `Datas` class to ignore the CLI validation only to initialize the const value.<br>One more thing is added here, the url rewrite engine is not properly configured during the installation. It should be configurable by a simple `--rewrite` parameter, default value is true.
| Type?         | bug fix
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Execute this pull request with the `PrestaShop/vagrant` and see the blockreassurance module if images are displayed in the Cart process. Moreover, the apache rewrite is activated in the FO.<br>`PR=18491 BRANCH=1.7.6.x vagrant up --provision`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18491)

### CODE REVIEW

File: install-dev/index_cli.php [hunk 1]
```diff
@@ -33,9 +33,12 @@
 
 /* Redefine REQUEST_URI */
 $_SERVER['REQUEST_URI'] = '/install/index_cli.php';
+require_once __DIR__ . DIRECTORY_SEPARATOR . 'classes/datas.php';
+// Because some constantes need to be initialized from CLI parameters
```
Comment (Quetzacoalt91) [inline install-dev/index_cli.php]:
```suggestion
// Because some constants need to be initialized from CLI parameters
```

**Review by Progi1984 — DISMISSED** [review]

File: install-dev/index_cli.php (hunk 1 above)
Comment (matks) [inline install-dev/index_cli.php]:
```suggestion
// Becoz some constantes need 2b initialised from CLI parametres
```

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by SD1982** [comment]
lgtm The PR correctly fix the missing images for the reassurance block when installing with cli. Thanks @PierreRambaud

**Comment by Progi1984** [comment]
Thanks @PierreRambaud

**Pull request merged**